apps/ledger-live-desktop/src/newArch/AnalyticsOptInPrompt/hooks/useVariantA.tsx

```diff
@@ -42,11 +60,47 @@ const useVariantA = ({ setPreventBackNavigation, goBackToMain, onSubmit }: UseVa
       dispatch(setShareAnalytics(AnalyticsData));
       dispatch(setSharePersonalizedRecommandations(PersonalizationData));
       onSubmit?.();
+      trackClick("Share", shouldWeTrack);
+    }
+  };
+
+  const handlePreferencesChange = (newPreferences: Record<FieldKeySwitch, boolean>) => {
+    const oldPreferences = preferences;
+    setPreferences(newPreferences);
+    for (const key in newPreferences) {
+      const fieldKey = key as FieldKeySwitch;
+      if (newPreferences[fieldKey] !== oldPreferences[fieldKey]) {
+        clickOnToggle(fieldKey, newPreferences[fieldKey]);
+      }
     }
   };
 
-  const handlePreferencesChange = (preferences: Record<FieldKeySwitch, boolean>) => {
-    setPreferences(preferences);
+  const trackClick = (button: string, shouldWeTrack: boolean) => {
+    track(
+      "button_clicked",
+      {
+        button,
+        variant,
+        flow,
+        page: "Analytics Opt In Prompt Main",
```

Comment:
It should be a parameter as well because you are calling the `trackClick` function from both `Main` and `Preferences` screens